tests/testthat/test_ksvm_kpar.R

```diff
@@ -0,0 +1,24 @@
+context("ksvm.kpar")
+
+test_kpar = function(kernel, kpar) {
+  learner = LearnerRegrKSVM$new()
+  learner$param_set$values = mlr3misc::insert_named(
+    learner$param_set$values,
+    append(list(kernel = kernel), kpar)
+  )
+  expect_learner(learner)
+  result = run_autotest(learner)
+  expect_true(result, info = result$error)
+
+  learner = LearnerClassifKSVM$new()
+  learner$param_set$values = mlr3misc::insert_named(
+    learner$param_set$values,
+    append(list(kernel = kernel), kpar)
+  )
+  expect_learner(learner)
+  result = run_autotest(learner)
+  expect_true(result, info = result$error)
+}
+
+test_kpar("rbfdot", list(sigma = 5))
+test_kpar("polydot", list(degree = 5, scale = 2, offset = 10))
```

Comment:
We should add at least one check per kpar with all allowed nested hyperpars.

Also there should be a test that non-allowed hyperpars cause an error.
You can write one test containing all sub-pars and apply it over all kpar values.
Since no kpar options accepts all sub-pars, there should always be the same error.

@mllg 

Currently the error message looks like

```r
 Error in (function (xs)  : 
  Assertion on 'xs' failed: Condition for 'offset' not ok: kernel anyof polydot, tanhdot; instead: kernel=rbfdot.
```

The wording sounds a bit suboptimal. Should we adjust this in paradox?

Reply:
I have added tests for all kernels and also added dependencies for nu, C and all of the hyperparams. I didnt yet add a test for non-allowed hyperparams. But while testing I noticed that some of the kernels dont work properly (see #8).